packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/explore/SQLEditor.tsx

```diff
@@ -39,44 +42,54 @@ export function SQLEditor({ table }: Props) {
     form.reset({ query });
   }, [query, form]);
 
+  const updateHeight = () => {
+    if (editorRef.current) {
+      const contentHeight = Math.min(200, editorRef.current.getContentHeight());
+      if (containerRef.current) {
+        containerRef.current.style.height = `${contentHeight}px`;
+      }
+
+      editorRef.current.layout({
+        width: editorRef.current.getLayoutInfo().width,
+        height: contentHeight,
+      });
+    }
+  };
+
   return (
     <Form {...form}>
       <form
-        className={cn(
-          "relative flex w-full flex-grow items-center justify-center bg-black align-middle",
-          "h-10 max-h-10 rounded-md border px-3 py-2 ring-offset-background",
-          {
-            "outline-none ring-2 ring-ring ring-offset-2": isFocused,
-          },
-        )}
+        className={cn("relative w-full rounded-md border bg-black px-3 py-2 ring-offset-background", {
+          "outline-none ring-2 ring-ring ring-offset-2": isFocused,
+        })}
         onSubmit={handleSubmit}
       >
         <FormField
           name="query"
           render={({ field }) => (
-            <Editor
-              width="100%"
-              height="21px"
-              theme="hc-black"
-              value={field.value}
-              options={monacoOptions}
-              language="sql"
-              onChange={(value) => field.onChange(value)}
-              onMount={(editor) => {
-                editor.onDidFocusEditorText(() => {
-                  setIsFocused(true);
-                });
+            <div ref={containerRef} className="min-h-[21px] w-full">
+              <Editor
+                width="100%"
+                theme="hc-black"
+                value={decodeURIComponent(field.value)}
+                options={monacoOptions}
+                language="sql"
+                onChange={(value) => field.onChange(encodeURIComponent(value ?? ""))}
+                onMount={(editor) => {
+                  editorRef.current = editor;
 
-                editor.onDidBlurEditorText(() => {
-                  setIsFocused(false);
-                });
-              }}
-              loading={null}
-            />
+                  updateHeight();
+                  editor.onDidContentSizeChange(updateHeight);
+                  editor.onDidFocusEditorText(() => setIsFocused(true));
+                  editor.onDidBlurEditorText(() => setIsFocused(false));
+                }}
+                loading={null}
+              />
+            </div>
           )}
         />
 
-        <Button className="absolute right-1 top-1 h-8 px-4" type="submit">
+        <Button className="absolute bottom-1 right-1 h-8 px-4" type="submit">
```

Comment:
no change needed here but just wanted to note that I've found using position relative/absolute to create stacking contexts ends up being more trouble than its worth, because you end up fighting z-index or weird layering of things like modals etc.

now that we have grids, I now tend to use an approach like
```
<div className="grid">
  <div className="col-start-1 row-start-1">
    ...
  </div>
  <div className="col-start-1 row-start-1">
    ...
  </div>
</div>
```

obviously you can get more nuanced with this where you define an actual grid with template rows/cols and allow the top layer to take up only as much room as needed, but I only reach for that when I need to be specific about placement/alignment/spacing

another thing to note about this strategy is that the top layer (if using a full-size layer) will take over as the click target, so also need to use something like `pointer-events-none` on the layer's container and `pointer-events-auto` in each child